Dockerfile

```diff
@@ -33,8 +33,13 @@ ENV PATH="/root/go/bin:${PATH}"
 
 # Git identifier to checkout
 ARG TNF_VERSION
-# Pull the required version of TNF
-RUN git clone --depth=1 --branch=${TNF_VERSION} https://github.com/test-network-function/test-network-function ${TNF_SRC_DIR}
+ARG TNF_SRC_URL=https://github.com/test-network-function/test-network-function
+ARG GIT_CHECKOUT_TARGET=$TNF_VERSION
+
+# Clone the TNF source repository and checkout the target branch/tag/commit
+RUN git clone --no-single-branch --depth=1 ${TNF_SRC_URL} ${TNF_SRC_DIR}
+RUN git -C ${TNF_SRC_DIR} fetch origin ${GIT_CHECKOUT_TARGET}
+RUN git -C ${TNF_SRC_DIR} checkout ${GIT_CHECKOUT_TARGET}
```

Comment:
I've tried different approaches until I finally settled on this one.

One alternative solution would be to create a boolean build argument called something like `TNF_VERSION_AS_COMMIT_SHA`, that would then set which `git clone` operation would be performed (there would be a separate git clone syntax for branches/tags, and a separate for cloning by SHA of a commit). Sadly, introducing general conditional logic in Dockerfiles is nontrivial, and one-line conditional expressions significantly reduce readability of Dockerfile instructions.

I think the proposed approach is a nice compromise. Still quite clean, and does what it should.